transform/transformers/base/base.py

```diff
@@ -0,0 +1,93 @@
+# pylint: disable=line-too-long, too-many-arguments), invalid-name
+"""Base transformer"""
+from abc import ABC, abstractmethod
+from typing import Tuple, Dict
+from pyspark.sql import DataFrame, SparkSession
+from pyspark.sql.types import StructType
+from transform.utils.join_dfs import create_df, join_different_dfs
+from transform.utils.utils import drop_columns, add_columns, replace_empty_str
+from transform.transformations.common import add_tg_fields
+
+
+class BaseTransformer(ABC):
+    """Base transformer class"""
+
+    def __init__(
+        self,
+        desired_type: str,
+        cols_to_add: Tuple,
+        cols_to_drop: Tuple,
+        cols_to_rename: Dict,
```

Comment:
If we're typing dict it's good to type what type is the key and value `dict[str, Any]`.